yfinance/base.py

```diff
@@ -405,6 +394,27 @@ def cleanup(data):
 
         self._fundamentals = True
 
+    def analyst_recommendations(self, data):
+        # test if the type of data is dict
+        if isinstance(data, dict):
```

Comment:
This should use `assert`

Reply:
Isn't assert for testing? We used assert in our testing. The change in base.py was our software fix.